src/dtclient/kubernetes_settings.go

```diff
@@ -6,21 +6,30 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"strconv"
 	"strings"
 
 	"github.com/pkg/errors"
 )
 
 type postKubernetesSettings struct {
-	Label                           string `json:"label"`
-	ClusterIdEnabled                bool   `json:"clusterIdEnabled"`
-	ClusterId                       string `json:"clusterId"`
-	CloudApplicationPipelineEnabled bool   `json:"cloudApplicationPipelineEnabled"`
-	OpenMetricsPipelineEnabled      bool   `json:"openMetricsPipelineEnabled"`
-	Enabled                         bool   `json:"enabled"`
-	EventProcessingActive           bool   `json:"eventProcessingActive"`
-	EventProcessingV2Active         bool   `json:"eventProcessingV2Active"`
-	FilterEvents                    bool   `json:"filterEvents"`
+	Label            string `json:"label"`
+	ClusterIdEnabled bool   `json:"clusterIdEnabled"`
+	ClusterId        string `json:"clusterId"`
+	Enabled          bool   `json:"enabled"`
+	*monitoringSettings
+}
+
+type monitoringSettings struct {
+	CloudApplicationPipelineEnabled bool `json:"cloudApplicationPipelineEnabled"`
+	OpenMetricsPipelineEnabled      bool `json:"openMetricsPipelineEnabled"`
+	EventProcessingActive           bool `json:"eventProcessingActive"`
+	EventProcessingV2Active         bool `json:"eventProcessingV2Active"`
+	FilterEvents                    bool `json:"filterEvents"`
+}
+
+func (d *postKubernetesSettings) UnmarshalJSON(b []byte) error {
```

Comment:
does this need to be its own method?

Reply:
Thanks for this point, I was able to remove it after renaming of field  monitoringSettings->MonitoringSettings
Actually the problem was in tests on 

var parsedBody []postKubernetesSettingsBody
err = json.Unmarshal(body, &parsedBody)

as monitoringSettings should be parsed as part of postKubernetesSettings, but I got an error so U added additional UnmarshalJSON. But after renaming monitoringSettings->MonitoringSettings it did work as exoected.